File: lab10/src/by/bntu/fitr/minigame/domain/east/Human.java
package by.bntu.fitr.minigame.domain.east;

import by.bntu.fitr.minigame.domain.Creature;
import by.bntu.fitr.minigame.domain.Role;
import by.bntu.fitr.minigame.domain.Weapon;

public class Human extends Creature {
    private Weapon weapon = Weapon.SWORD;
    private Role role = Role.WARRIOR;

    public Human(int health, int strength, int agility, int intelligence) {
        super(health, strength, agility, intelligence);
    }

    @Override
    protected int getWeaponDamage() {
        return weapon.getDamage();
    }

    @Override
    public int buff() {
        return role.getPercent() / 100 * getStrength() + role.getPercent() / 100 * getHealth();
    }

    @Override
    public String toString() {
        return "Human{" + super.toString() +
                "weapon=" + weapon +
                ", role=" + role +
                '}';
    }
}

File: lab10kt/src/main/kotlin/by/bntu/fitr/povt/korotkevich/model/west/Human.kt
package by.bntu.fitr.povt.korotkevich.model.west

import by.bntu.fitr.povt.korotkevich.model.Creature
import by.bntu.fitr.povt.korotkevich.model.Role
import by.bntu.fitr.povt.korotkevich.model.Weapon
import by.bntu.fitr.povt.korotkevich.model.buff

class Human(override val health: Int = 100, override val strength: Int = 30,
            override val agility: Int = 20, override val intelligence: Int = 20,
            val role: Role = Role.KNIGHT, val weapon: Weapon = Weapon.SWORD) : Creature() {

    override val id: String = "Human"

    override fun getSelfDamage() = weapon.damage + buff(role)

    override fun toString() = StringBuilder("Human(").apply {
        append("${super.toString()}, role: $role, weapon: $weapon)")
    }.toString()
}